src/app/views/authentication/Authentication.tsx

```diff
@@ -0,0 +1,142 @@
+      this.setState({
+        authenticated: JSON.parse(authenticated),
+      });
+    }
+  }
+
+  public signIn = () => {
+    this.initialiseAuthentication();
+    hello('msft').login({
+        response_type: 'token',
+        scope: DEFAULT_USER_SCOPES,
+        display: 'popup',
+      });
+  };
+
+  private initialiseAuthentication() {
+    const { actions, queryActions } = this.props;
+    hello.init({
+      msft: {
+        oauth: {
+          version: 2,
+          auth: AUTH_URL,
+          grant: TOKEN_URL,
+        },
+        scope_delim: ' ',
+        form: false,
+      },
+      msft_admin_consent: {
+        oauth: {
+          version: 2,
+          auth: ADMIN_AUTH_URL,
+          grant: TOKEN_URL,
+        },
+        scope_delim: ' ',
+        form: false,
+      },
+    } as any);
+    hello.init({
+      msft: 'cb2d7367-7429-41c6-ab18-6ecb336139a6',
+      msft_admin_consent: 'cb2d7367-7429-41c6-ab18-6ecb336139a6',
+    }, {
+        redirect_uri: window.location.pathname,
+      });
+    hello.on('auth.login', async (auth) => {
+        let accessToken;
+        const authenticated = {...this.state.authenticated};
+        if (auth.network === 'msft') {
+          const authResponse = hello('msft').getAuthResponse();
+          accessToken = authResponse.access_token;
+          if (accessToken) {
+            authenticated.token = accessToken;
+          }
+          authenticated.status = true;
+          if (actions) {
+            actions.authenticateUser(authenticated);
+          }
+        }
+        if (accessToken) {
+          try {
+            const userInfo = (queryActions) ? await queryActions.runQuery(USER_INFO_URL) : null;
+            const jsonUserInfo = userInfo.response.body;
+            authenticated.user = {...{},
+                                  displayName: jsonUserInfo.displayName,
+                                  emailAddress: jsonUserInfo.mail || jsonUserInfo.userPrincipalName,
+            };
+            if (actions) {
+              actions.authenticateUser(authenticated);
+              this.setState({
+                authenticated,
+              });
+            }
+          } catch (e) {
+            // tslint:disable-next-line:no-console
+            console.log(e);
+          }
+        }
+      });
+  }
+
+  public render() {
+    const { authenticated } = this.state;
+    return (
+      <div className='authentication-container'>
+        <PrimaryButton onClick={this.signIn} className='signIn-button'>
+          Sign In
+        </PrimaryButton>
+        <div className='authentication-details'>
+          <span className='user-name'>{authenticated.user.displayName}</span>
+          <br />
+          <span className='user-email'>{authenticated.user.emailAddress}</span>
+        </div>
+      </div>
+    );
+  }
+}
+
+function mapDispatchToProps(dispatch: Dispatch): object {
+  return {
+    actions: bindActionCreators(authActionCreators, dispatch),
```

Comment:
You don't have to call `bindActionCreators` twice. You can do the following
```
bindActionCreators({ queryActions, authActions }, dispatch)
```

Reply:
awesome